maliciousscan_test.go

```diff
@@ -0,0 +1,168 @@
+package main
```

Comment:
you can remove some of the funcs here and combine them to remove code duplication.
take that into considirations